solidity/contracts/bridge/Bridge.sol

```diff
@@ -794,10 +955,577 @@ contract Bridge is Ownable {
+            //       `extractValueAt` and `extractHashAt` in `bitcoin-spv-sol`
+            //       in order to avoid allocating bytes in memory.
+            uint256 outputLength =
+                redemptionTxOutputVector.determineOutputLengthAt(
+                    outputStartingIndex
+                );
+            bytes memory output =
+                redemptionTxOutputVector.slice(
+                    outputStartingIndex,
+                    outputLength
+                );
+
+            // Extract the value and hash from given output.
+            uint64 outputValue = output.extractValue();
+            bytes memory outputHash = output.extractHash();
+
+            if (
+                info.changeValue == 0 &&
+                keccak256(outputHash) == walletPubKeyHashKeccak &&
+                outputValue > 0
+            ) {
+                // If we entered here, that means the change output with a
+                // proper non-zero value was found.
+                info.changeIndex = uint32(i);
+                info.changeValue = outputValue;
+            } else {
+                // If we entered here, that the means the given output is
+                // supposed to represent a redemption. Build the redemption key
+                // to perform that check.
+                uint256 redemptionKey =
+                    uint256(
+                        keccak256(
+                            abi.encodePacked(walletPubKeyHash, outputHash)
+                        )
+                    );
+
+                if (
+                    pendingRedemptions[redemptionKey].requestedAt != uint32(0)
+                ) {
+                    // If we entered here, that means the output was identified
+                    // as a pending redemption request.
+                    RedemptionRequest storage request =
+                        pendingRedemptions[redemptionKey];
+                    // Compute the request's redeemable amount as the requested
+                    // amount reduced by the treasury fee. The request's
+                    // minimal amount is then the redeemable amount reduced by
+                    // the maximum transaction fee.
+                    uint64 redeemableAmount =
+                        request.requestedAmount - request.treasuryFee;
+                    // Output value must fit between the request's redeemable
+                    // and minimal amounts to be deemed valid.
+                    require(
+                        redeemableAmount >= outputValue &&
+                            outputValue >= redeemableAmount - request.txMaxFee,
+                        "Output value is not acceptable for redemption request"
+                    );
+                    // Add the redeemable amount to the total burnable value
+                    // the Bridge will use to decrease its balance in the Bank.
+                    info.totalBurnableValue += redeemableAmount;
```

Comment:
I think the current version is correct and has nothing to do with `request.txMaxFee` which is only used to check the lowe bound. This is because the `outputValue` and the fee share for that output always sum up to the `redeemableAmount`. Suppose a request like that:
```
requestedValue: 5 BTC
treasuryFee: 0.5 BTC
txMaxFee: 0.3
```
We have `redeemableAmount = 5 BTC - 0.5 BTC = 4.5 BTC`. The minimal acceptable amount is `redeemableAmount - txMaxFee = 4.5 BTC - 0.3 BTC = 4.2 BTC`. To be accepted, an output must have a value `x` where `4.2 BTC <= x <= 4.5 BTC`. Let's say there is only one output in the transaction. There is a single input having `4.5 BTC`. We have, for example, following possibilities:
- fee is `0.4 BTC`, redeemer obtains `4.1 BTC` ->  this sums up to `4.5 BTC` of the redeemable amount though it will revert due to being below the minimum
- fee is `0.3 BTC`, redeemer obtains `4.2 BTC` -> this also sums up to `4.5 BTC`
- fee is `0.1 BTC`, redeemer obtains `4.4 BTC` -> this also sums up to `4.5 BTC`
- fee is `0 BTC` (not possible), redeemer obtains `4.5 BTC` -> this also sums to `4.5 BTC`

The same occurs for multiple outputs transactions. The wallet must just construct the transaction in the way it respects max tx fee per output.